### PR TITLE
PLNSRVCE-769: Add minio tenant for tekton results

### PR DESCRIPTION
**Pull request by gabemontero**
Trying to help out @AndrienkoAleksandr propagate #419 while he has electricity issues

Simply rebased off of main and squashed the commits.

Let's see how CI does here.

@adambkaplan @Roming22 FYI

### CODE REVIEW

**Comment by gabemontero** [comment]
I'll start working on the shellcheck / yamllint items

**Comment by gabemontero** [comment]
static checks clean

**Comment by gabemontero** [comment]
hmmm can't seem to assign @AndrienkoAleksandr nor add him as a Reviewer

File: operator/gitops/argocd/pipeline-service.yaml
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service.yaml:14]: will change and see what the result is

File: developer/config.yaml
Comment (gabemontero) [inline developer/config.yaml:9]: found another one @adambkaplan

Comment (gabemontero) [reply developer/config.yaml]: update pushed

**Comment by gabemontero** [comment]
have a debug commit up based on the full logs @Roming22 provided me (he's called out to someone else to see if they can fix my ci cluster auth issues)

**Comment by gabemontero** [comment]
Well the fact we can't dump events from these scripts should be adjusted for debug down the road:
```
- pod with label app=minio: ........................................Error from server (Forbidden): events is forbidden: User "system:serviceaccount:pipeline-service:pipeline-service-manager" cannot list resource "events" in API group "" in the namespace "tekton-results"
command terminated with exit code 1
```

I'll see about getting the rest of the logs tomorrow either with my hopefully fixed ID on the test cluster, or getting somebody to pastebin again.

**Comment by gabemontero** [comment]
Looks like @AndrienkoAleksandr 's work around from https://github.com/openshift-pipelines/pipeline-service/pull/419#issuecomment-1386323166
did not work in CI:

```
  2m37s       Warning   FailedCreate           statefulset/storage-pool-0                 create Pod storage-pool-0-0 in StatefulSet storage-pool-0 failed error: pods "storage-pool-0-0" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.containers[0].securityContext.runAsUser: Invalid value: 1000610001: must be in the ranges: [1000630000, 1000639999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]
```

I'll see about bumping the minio level, pick up that change that got merged up there.

That way, we can leverage openshift's randomized setting of the UID.

Other various items were also noted as forbidden in that event.  Looks like we'll need to associated an SCC with some additional permissions to get that Pod's creation to pass.

@adambkaplan FYI ^^

**Comment by gabemontero** [comment]
> Looks like @AndrienkoAleksandr 's work around from [#419 (comment)](https://github.com/openshift-pipelines/pipeline-service/pull/419#issuecomment-1386323166) did not work in CI:
> 
> ```
>   2m37s       Warning   FailedCreate           statefulset/storage-pool-0                 create Pod storage-pool-0-0 in StatefulSet storage-pool-0 failed error: pods "storage-pool-0-0" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.containers[0].securityContext.runAsUser: Invalid value: 1000610001: must be in the ranges: [1000630000, 1000639999], provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "nonroot": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]
> ```
> 
> I'll see about bumping the minio level, pick up that change that got merged up there.
> 
> That way, we can leverage openshift's randomized setting of the UID.
> 
> Other various items were also noted as forbidden in that event. Looks like we'll need to associated an SCC with some additional permissions to get that Pod's creation to pass, but I'll fix the UID first, then see.
> 
> @adambkaplan FYI ^^

UPDATE - only 4.5.7 is available through the normal channel.  At least for now you have to pay for 4.5.8 (which has the PR 1403 change), per when I try to install from the console on my personal 4.11 cluster for the day.

Conversely, @AndrienkoAleksandr alternative minio PR has not been merged.

IIRC the uid range itself may be somewhat random, but I'll try something withing `[1000630000, 1000639999]` from the event and see what happens.

Otherwise, we may have to bump permissions on this thing beyond non-root.

**Comment by gabemontero** [comment]
OK we have passing tests @adambkaplan @Roming22 @AndrienkoAleksandr !!

I'm going to squash a couple of commits specific to minio, and then generalize my debug additions a bit, and then post here about final reviews / merging.

I suspect minimally we'll want more testing perhaps, but first blush I suggest doing that in follow up PRs, as this one and @AndrienkoAleksandr https://github.com/openshift-pipelines/pipeline-service/pull/419 have been the lucky recipients of multiple rebases with other PRs merging ahead.

**Comment by gabemontero** [comment]
OK this is ready for review @adambkaplan @Roming22 @AndrienkoAleksandr

**Review by adambkaplan — APPROVED** [review]
Generally looks good. I ask that the debug commits be squashed into a single one so it's clearer that we're adding RBAC to debug failing tests.

**Comment by gabemontero** [comment]
> Generally looks good. I ask that the debug commits be squashed into a single one so it's clearer that we're adding RBAC to debug failing tests.

thanks and done @adambkaplan

**Comment by AndrienkoAleksandr** [comment]
Hello. Sorry, I think workaround with security context was wrong. If I understood docs correctly, Openshift provides different userId and groupId. So manually provided group and user id can work on the one cluster and it won't work in the another. Looks like we need to provide more complex security context stuff... But I need to read more docs and play with this staff to figure out how to provide more universal solution... 
Ideally and the simplier it would be nice if we provide empty security context and Openshift generate default one, but minio operator broke this stuff https://github.com/minio/operator/pull/1403 . And looks like my proposal to turn it back stuck https://github.com/minio/operator/pull/1407 ...

**Comment by Roming22** [comment]
With @AndrienkoAleksandr latest comment, I'm going to put a "WIP" in the PR description so we know not to merge it at the moment.

**Comment by gabemontero** [comment]
> @gabemontero Hello. Thanks a lot for help. But, sorry, I think my workaround with security context was wrong. If I understood docs correctly, Openshift provides different range for userId and groupId. So manually provided group and user id can work on the one cluster and it won't work in the another. Looks like we need to provide more complex security

Yeah I think you may be correct @AndrienkoAleksandr, and that ties in with the suspicion I noted in https://github.com/openshift-pipelines/pipeline-service/pull/443#issuecomment-1397393459

So I'll engage in a couple of items for this, short term / today
0) bring up the operator catalog today, confirm if 4.5.7 is still the version in the stable channel (which does not have the upstream 1403 is in), or if it has gone up to 4.5.8 (which does have the 1403 change)
1) then locally, simply comment out the ID override, and confirm the behavior at the version currently in the stable channel
2) if need be,  based on whatever the behavior / version is, look to see if escalated permissions for the tenant can get us past any hacks like this;  i.e. I'll create a specific serviceaccount, add more privileged SCCs to it, possible turn off the non-root bit / add the privileged bit, etc.  i.e. create a short term work around while we sort things out upstream 

> context stuff... But I need to read more docs and play with this staff to figure out how to provide more universal solution... Ideally and the simplier it would be nice if we provide empty security context and Openshift generate default one, but minio operator broke this stuff [minio/operator#1403](https://github.com/minio/operator/pull/1403) . And looks like my proposal to turn it back stuck [minio/operator#1407](https://github.com/minio/operator/pull/1407) ...

Yeah if 1403/4.5.8 is in fact not addressing things, then we push on your PR 1407, given them the errors we see on 4.11 with 4.5.8, provide yamls so that they can reproduce, perhaps given them instructions for bringing up a 4.11 cluster through OKD or one of our free tiers at cloud.openshift.com if that is possible, whatever.

Now, to "officially" determine the state of 4.5.8, until it lands in stable channel, I'm not sure how we test it in our CI here.  But maybe there is some operator hub nuance I'm missing that @adambkaplan may  be aware of (he has messed with that a lot more than I).

@Roming22 FYI and based on where the findings land, if we have an acceptable work around for this pod permission item, I'll remove the WIP tag and we'll go from there.

thanks everyone

**Comment by gabemontero** [comment]
hmmm .... now it is failing in the `"- Setup working directory:"` step in dev_setup.sh before we even get to the tenant verification:

```
[plnsvc-setup : run-plnsvc-setup]   - Generating shared manifests:
[plnsvc-setup : run-plnsvc-setup]     - tekton-chains manifest:
[plnsvc-setup : run-plnsvc-setup]       + chains_kustomize=/tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/kustomization.yaml
[plnsvc-setup : run-plnsvc-setup]       + chains_namespace=/tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/namespace.yaml
[plnsvc-setup : run-plnsvc-setup]       + chains_secret=/tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/signing-secrets.yaml
[plnsvc-setup : run-plnsvc-setup]       + '[' '!' -e /tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/kustomization.yaml ']'
[plnsvc-setup : run-plnsvc-setup]       ++ dirname /tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/kustomization.yaml
[plnsvc-setup : run-plnsvc-setup]       + chains_tmp_dir=/tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/tmp
[plnsvc-setup : run-plnsvc-setup]       + mkdir -p /tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/tmp
[plnsvc-setup : run-plnsvc-setup]       ++ head -c 12 /dev/urandom
[plnsvc-setup : run-plnsvc-setup]       ++ base64
[plnsvc-setup : run-plnsvc-setup]       + cosign_passwd=PASiWB5j53bH1mS5
[plnsvc-setup : run-plnsvc-setup]       + echo -n PASiWB5j53bH1mS5
[plnsvc-setup : run-plnsvc-setup]       + cosign_image=quay.io/redhat-appstudio/appstudio-utils:eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b
[plnsvc-setup : run-plnsvc-setup]       + podman run --rm --env COSIGN_PASSWORD=PASiWB5j53bH1mS5 --volume /tmp/tmp.9tCe0PagE2/credentials/manifests/compute/tekton-chains/tmp:/workspace:z --workdir /workspace --entrypoint /usr/bin/cosign quay.io/redhat-appstudio/appstudio-utils:eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b generate-key-pair
[plnsvc-setup : run-plnsvc-setup]       Error: crun: writing file `/proc/994/setgroups`: Operation not permitted: OCI permission denied
[plnsvc-setup : run-plnsvc-setup] command terminated with exit code 126
```

there has been 1 commit since yesterday, though I'm not sure if it has any bearing here

sidetracked looking into this after lunch

**Comment by gabemontero** [comment]
the error from https://github.com/openshift-pipelines/pipeline-service/pull/443#issuecomment-1398723789 is while setting up the tekton chains manifest using quay.io/redhat-appstudio/appstudio-utils:eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b

that line of code from that ref is 2 months old, and the image itself is 10 months old

`quay.io/redhat-appstudio/appstudio-utils                  eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b   08ec8eb4edf3   10 months ago   537MB`

that image was updated as recently as 6 hours ago

feels like an unrelated upstream dependency going bad, but I'll try a master branch test PR to see

**Comment by gabemontero** [comment]
confirmed ... seeing the same failure in https://github.com/openshift-pipelines/pipeline-service/pull/447

also, see now that 4.5.8 of minio is in the stable channel when installing from the openshift console

**Comment by gabemontero** [comment]
> confirmed ... seeing the same failure in #447
> 
> also, see now that 4.5.8 of minio is in the stable channel when installing from the openshift console

https://github.com/openshift-pipelines/pipeline-service/pull/449 should fix the regression

**Comment by gabemontero** [comment]
> confirmed ... seeing the same failure in #447
> 
> also, see now that 4.5.8 of minio is in the stable channel when installing from the openshift console

*HOWEVER* the minio that is getting installed in the ci run in much older:

`  installedCSV: minio-operator.v4.5.4`

So we need to see about fixing that.  I shut down my cluster for the day, but when I pick back up, I'll compare the subscription yamls, and see if I can find the magic bit that bumps up to 4.5.8.  Hopefully it is not an openshift version thing.

Also, while waiting for test runs, I found 

https://github.com/minio/operator/blob/master/docs/pod-security-policy.md

and 

https://github.com/minio/operator/blob/master/examples/kustomization/tenant-PodSecurityPolicies/minio-sa.yaml

Now, the old / now deprecated `PodSecurityPolicy` is very much like OpenShift `SecurityContextConstraints` (which still exist / are supported).   After getting to 4.5.8 and sorting out the deal with the uid/gid work around, depending on how that lands and a non-optimal situation exists, I'll most likely next look to creating an analogous SCC, a new SA for the tenant (vs. its use of the default SA if I read it right), and the necessary role bindings and hopefully see it work there.

@adambkaplan @AndrienkoAleksandr FYI ^^

**Comment by adambkaplan** [comment]
Something to be aware of is what kind of cluster we are spinning up for tests. We've seen issues where hypershift clusters don't have their catalogs updated, which might lead to what Gabe observed with the outdated operator version being installed.

**Comment by gabemontero** [comment]
On Fri, Jan 20, 2023 at 5:47 PM Adam Kaplan ***@***.***>
wrote:

> Something to be aware of is what kind of cluster we are spinning up for
> tests. We've seen issues where hypershift clusters don't have their
> catalogs updated, which might lead to what Gabe observed with the outdated
> operator version being installed.
>
These are definitely hypershit clusters we are deploying to.

Were there any work arounds or paths to fix that you recall Adam ?

**Comment by gabemontero** [comment]
On Fri, Jan 20, 2023 at 5:52 PM Gabe Montero ***@***.***> wrote:

>
>
> On Fri, Jan 20, 2023 at 5:47 PM Adam Kaplan ***@***.***>
> wrote:
>
>> Something to be aware of is what kind of cluster we are spinning up for
>> tests. We've seen issues where hypershift clusters don't have their
>> catalogs updated, which might lead to what Gabe observed with the outdated
>> operator version being installed.
>>
> These are definitely hypershit clusters we are deploying to.
>
> Were there any work arounds or paths to fix that you recall Adam ?
>

They do seem to have a kustomize install path.  If there are no solutions
to the operator based install on hypershift we could conceivably do that,
at least in the short term.

**Comment by gabemontero** [comment]
OK @AndrienkoAleksandr @adambkaplan @Roming22 we are getting passing tests using 4.5.8 minio installed via helm, where I do not have to set any uid/gid in the tenant security context.

Next steps:
- fix the yamllint complaints (may have some yamllint vs. helm conflicts in formatting and I may have to excluded the new helm chart sub dir from the yamllint check)
- once https://github.com/openshift-pipelines/pipeline-service/pull/451 merges I'll quit installing `helm` in this PR
- some commit cleanup, removal of the WIP designation, and final review / merge.

**Comment by gabemontero** [comment]
> OK @AndrienkoAleksandr @adambkaplan @Roming22 we are getting passing tests using 4.5.8 minio installed via helm, where I do not have to set any uid/gid in the tenant security context.
> 
> Next steps:
> 
>     * cross reference results using the nonroot SCC and the older minio version available on the hypershift clusters; if we get a fail, we move onto helm; if it succeeds, we decide direction either here in the PR or in slack/meetings
> 
>     * fix the yamllint complaints (may have some yamllint vs. helm conflicts in formatting and I may have to excluded the new helm chart sub dir from the yamllint check)
> 
>     * once [add helm to allow for helm based install of the minio operator #451](https://github.com/openshift-pipelines/pipeline-service/pull/451) merges I'll quit installing `helm` in this PR
> 
>     * some commit cleanup, removal of the WIP designation, and final review / merge.

Yep even at v4.5.4 of the minio operator (vs. v4.5.8) the `nonroot` SCC allows the tenant stateful set to come up @AndrienkoAleksandr @adambkaplan 

So we just need to decide if we are fine being stuck at the older version of the operator until the hypershift operator catalogs get updated, or if we want the have the flexibility of controlling the version we pick via the helm install method.

**Comment by AndrienkoAleksandr** [comment]
+1  to control operator version. It will save us from unepected bugs.

**Comment by gabemontero** [comment]
fyi @adambkaplan @AndrienkoAleksandr and myself finally had an opportunity to have a voice to voice conversation.  Net:
- moving off helm, we we can survive with older minio version for now
- this PR will not set anything security context related, and establish use of SCC and a dedicated SA for the tenant

Assuming CI passes, I'll remove the WIP designation, make sure @Roming22 and team have no more comments, and we'll merge.

@AndrienkoAleksandr can than pursue any tuning / post start up work arounds needed in follow up PRs.

**Comment by gabemontero** [comment]
OK all green

@Roming22 - from api team perspective we are good here per https://github.com/openshift-pipelines/pipeline-service/pull/443#issuecomment-1405084581; do you want one more round of infra team review, or can one of us hit the merge button?

**Comment by adambkaplan** [comment]
@gabemontero perhaps squash commits, then I'll let @Roming22 review so we're in sync.

**Comment by gabemontero** [comment]
> @gabemontero perhaps squash commits, then I'll let @Roming22 review so we're in sync.

commits squashed @adambkaplan

File: operator/gitops/argocd/pipeline-service/kustomization.yaml
Comment (adambkaplan) [inline operator/gitops/argocd/pipeline-service/kustomization.yaml:9]: @Roming22 I think this may present a problem when we try to get the change applied to infra-deployments. We may need to discuss with the core team or the GitOps team on how we manage overlays for AWS services.

Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/kustomization.yaml:9]: good catch @adambkaplan / I would agree

this stems from @AndrienkoAleksandr 's original commit, so we may want to cross verify with him in the end, but at first glance, given the moves he made into the base subdir, feels like this should be `tekton-results/base` and then the credential patches needed for the developer flow should be called directly from the ci-runner flow

infra-deployments definitely has the pattern of accessing base dirs ... see https://github.com/redhat-appstudio/infra-deployments/blob/main/components/jvm-build-service/kustomization.yaml#L3 for example

I'll see if I can do that, first in a separate commit, and we get clean CI,, then go from there

with all that, I'm certainly do not disagree wrt getting input from core/gitops in conjunction

File: operator/images/access-setup/content/bin/setup_work_dir.sh
Comment (Roming22) [inline operator/images/access-setup/content/bin/setup_work_dir.sh:203]: We avoid inlining yaml files as it prevents linting. Have a template that you modify on the fly, preferably using `yq`.

File: operator/images/cluster-setup/content/bin/install.sh
Comment (Roming22) [inline operator/images/cluster-setup/content/bin/install.sh:26]: If possible please call it `PROJECT_DIR`.

File: developer/openshift/operators/minio/tenant/tenant.yaml
Comment (adambkaplan) [inline developer/openshift/operators/minio/tenant/tenant.yaml:6]: Does it make more sense to make the `Tentant` a dev overlay?

**Review by Roming22 — COMMENTED** [review]
It's unclear to me what is the role of `operator/gitops/argocd/pipeline-service/tekton-results/overlays/dev`.

**Comment by gabemontero** [comment]
ok I've attempted to address, though I'm not crazy confident, @adambkaplan 's https://github.com/openshift-pipelines/pipeline-service/pull/443#discussion_r1088310506

but given @Roming22 's additional comments, where some of those are getting even more into @AndrienkoAleksandr 's original changes/commits from #419 (which had received prior reviews from various folks) that I just pulled over, and with the new sprint and my new assignment there, plus the openshift work I have to pick up during this sprint's timeframe, it is beginning to feel like my offer to assist @AndrienkoAleksandr may have reached the end of its viability.

either he takes this back over full time, assuming he has the cycles, or we apply other resources to help, or something else

but I'll report back one more time today after the current CI run completes, and we'll go from there.

File: operator/gitops/argocd/pipeline-service/kustomization.yaml
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/kustomization.yaml]: btw looking at the time stamps of the comments, it looks like @adambkaplan 's original comment here was edited after my response (54 minutes for his vs. 37 for mine as of this comment).

I think that may be how our duplicate refs to `/base` arose :-)

**Comment by gabemontero** [comment]
> It's unclear to me what is the role of `operator/gitops/argocd/pipeline-service/tekton-results/overlays/dev`.

yeah that is a @AndrienkoAleksandr question to answer

File: operator/docs/sre/examples/providers/credentials/secrets/bitwarden.yaml
Comment (gabemontero) [reply operator/docs/sre/examples/providers/credentials/secrets/bitwarden.yaml:4]: eesh ... I'll undo that here, but fyi @Roming22 this also existed in https://github.com/openshift-pipelines/pipeline-service/pull/419

File: operator/docs/sre/examples/providers/credentials/secrets/bitwarden.yaml
Comment (gabemontero) [reply operator/docs/sre/examples/providers/credentials/secrets/bitwarden.yaml:7]: eesh ... I'll undo that here, but fyi @Roming22 this also existed in #419

File: operator/gitops/sre/credentials/secrets/README.md
Comment (gabemontero) [reply operator/gitops/sre/credentials/secrets/README.md:24]: eesh ... I'll undo that here, but fyi @Roming22 this also existed in #419

**Comment by AndrienkoAleksandr** [comment]
> It's unclear to me what is the role of operator/gitops/argocd/pipeline-service/tekton-results/overlays/dev

We wanted to provide minio only for dev flow. For production we wanted to use amazon S3 log storage. So for me it was natural to split all minio related changes and move it to dev folder. I thought that we will create one more overlay folder in the another staging cluster project, where is will be described aws S3 storage secret. If I misunderstood how it should be, you can simply revert overlays for tekton-results.

File: developer/openshift/dev_setup.sh
Comment (AndrienkoAleksandr) [reply developer/openshift/dev_setup.sh:207]: I agree about revertation, but I found that in this case "| indent 2" breaks progressor with dots. You know, when we have long term operation we want to see dots "Minio: .....Exists". But this indention hide all dots and they prints only when operation was completed. @Roming22 shall we create an issue about that?

File: developer/openshift/dev_setup.sh
Comment (AndrienkoAleksandr) [reply developer/openshift/dev_setup.sh:221]: +1 in this place.

**Comment by gabemontero** [comment]
My attempt to move the application of `overlays/dev` did not work:

```
Name: "tekton-results", Namespace: "tekton-results"
from server for: "/source/operator/gitops/argocd/pipeline-service/tekton-results/overlays/dev": routes.route.openshift.io "tekton-results" is forbidden: User "system:serviceaccount:pipeline-service:pipeline-service-manager" cannot get resource "routes" in API group "route.openshift.io" in the namespace "tekton-results"
command terminated with exit code 1
```

I'll try just reverting the folder altogether per @AndrienkoAleksandr  comment ^^ and we can see where we are at tommorrow AM.

File: operator/gitops/argocd/pipeline-service/kustomization.yaml
Comment (gabemontero) [reply operator/gitops/argocd/pipeline-service/kustomization.yaml]: switching to base and removing overlays/dev is passing the ci tests

marking resolved

File: DEPENDENCIES.md
Comment (gabemontero) [reply DEPENDENCIES.md:46]: part of next push

File: DEPENDENCIES.md
Comment (gabemontero) [reply DEPENDENCIES.md:48]: part of next push

File: developer/config.yaml
Comment (gabemontero) [reply developer/config.yaml:23]: part of next push

File: developer/openshift/dev_setup.sh
Comment (gabemontero) [reply developer/openshift/dev_setup.sh:221]: part of next push

File: operator/images/access-setup/content/bin/setup_work_dir.sh
Comment (gabemontero) [reply operator/images/access-setup/content/bin/setup_work_dir.sh:189]: again I'll need to defer to @AndrienkoAleksandr to respond and comment on his original intent here

**Comment by gabemontero** [comment]
> My attempt to move the application of `overlays/dev` did not work:
> 
> ```
> Name: "tekton-results", Namespace: "tekton-results"
> from server for: "/source/operator/gitops/argocd/pipeline-service/tekton-results/overlays/dev": routes.route.openshift.io "tekton-results" is forbidden: User "system:serviceaccount:pipeline-service:pipeline-service-manager" cannot get resource "routes" in API group "route.openshift.io" in the namespace "tekton-results"
> command terminated with exit code 1
> ```
> 
> I'll try just reverting the folder altogether per @AndrienkoAleksandr comment ^^ and we can see where we are at tommorrow AM.

So reverting of `overlay/devs`, along with reverting the bitwarden related changes, still passes CI, meaning the tenant pod at least comes up.

I'm about to push some of the minor of @Roming22 comments.

On a couple of @Roming22 's questions, I have had to defer to @AndrienkoAleksandr to explain his original intent.  I made comments in those threads when that was needed.

The in-line yaml and renaming a bash env var are an uptick up, I'll address those if time permits today while I get back to my official assignments.

File: developer/openshift/dev_setup.sh
Comment (Roming22) [reply developer/openshift/dev_setup.sh]: Yes, create an issue.

**Comment by Roming22** [comment]
@AndrienkoAleksandr If minio is for dev only, and is used to satisfy a pre-requisite to deploy pipeline-service, then the manifests should be in `developer/openshift/operators`, and deployed before deploying pipeline-service. If minio is used as fallback during deployment if another storage backend is not not found, then it can stay where it is.

Your comment on adding an overlay for the staging cluster is worrisome. You are mistaking this repository for a gitops repository managing pipeline-service clusters, which it is not. This repository should not be aware of any cluster operating pipeline-service. Any configuration related to the staging cluster should go in infra-deployments.

File: developer/openshift/dev_setup.sh
Comment (AndrienkoAleksandr) [reply developer/openshift/dev_setup.sh:191]: @Roming22 I think if user deployed pipeline service using dev_setup.sh script and user didn't set up values in the config.yaml, then defaults will come from here: https://github.com/openshift-pipelines/pipeline-service/pull/443/files#diff-f394b1c9744cc285b6ac80384e967bd73210dd72a905e1ad7581b8ecab4e3773R101

File: operator/images/access-setup/content/bin/setup_work_dir.sh
Comment (AndrienkoAleksandr) [reply operator/images/access-setup/content/bin/setup_work_dir.sh:189]: @gabemontero remove this line, please.

**Comment by AndrienkoAleksandr** [comment]
> Your comment on adding an overlay for the staging cluster is worrisome. You are mistaking this repository for a gitops repository managing pipeline-service clusters, which it is not. This repository should not be aware of any cluster operating pipeline-service. Any configuration related to the staging cluster should go in infra-deployments.

For staging cluster configuration we have separated issue.  I only wanted to describe flow in general - what we want to see in the end after some amount of the pull requests. For dev purpose we want to have minio, for staging we want to use another S3 storage.

File: operator/images/access-setup/content/bin/setup_work_dir.sh
Comment (gabemontero) [reply operator/images/access-setup/content/bin/setup_work_dir.sh]: removed / pushed

**Review by Roming22 — APPROVED** [review]

**Pull request merged**
